Add new email pattern categories and load them dynamically from block patterns

lysyjan opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

 This PR adds category tabs navigation to the email template selection modal, allowing users to filter templates by category.

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->
This PR adds category tabs navigation to the email template selection modal, improving the user experience when browsing and selecting email templates.

  Key changes:
  - Added dynamic category tabs to the template selection modal that are populated from block pattern categories
  - Categories are now loaded dynamically from registered block patterns instead of being hardcoded
  - Added new email pattern categories: Newsletter, Welcome, and Abandoned Cart
  - Template category is now derived from the first category of its content pattern
  - Localized "Recent" category label for internationalization support

  Related to WOOPRD-1118.

### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

| Before | After |
| ------ | ----- |
|<img width="1679" height="1211" alt="Screenshot 2025-12-12 at 20 49 08" src="https://github.com/user-attachments/assets/15c78c5a-ee1d-4e80-94c6-13763f8bebd9" />|<img width="1678" height="1216" alt="Screenshot 2025-12-12 at 18 56 07" src="https://github.com/user-attachments/assets/7bedba3d-e82b-4cf8-a424-b233e438fad7" />|



<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Use those new packages in MailPoet
2. Visit MailPoet > Newsletters page
3. Click on the button for creating a new email in the block email editor 

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Milestone

<!-- DO NOT remove or modify this section (other than to check the box). -->

<!-- milestone-target-selection -->
- [ ] Automatically assign milestone for the **[next WooCommerce version](../blob/trunk/plugins/woocommerce/woocommerce.php#L6)**
<!-- /milestone-target-selection -->

> **Note:** Check the box above to have the milestone automatically assigned when merged.
> Alternatively (e.g. for point releases), manually assign the appropriate milestone.


### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

lysyjan committed:
Localized template category labels in email editor

lysyjan committed:
Updated template category logic to dynamically use first category from content pattern

lysyjan committed:
Added new email pattern categories: Basic, Welcome, and Abandoned Cart

lysyjan committed:
Load template categories dynamically from block patterns

lysyjan committed:
Renamed "Basic" email pattern category to "Newsletter"

lysyjan committed:
Add category tabs navigation and method to retrieve email pattern categories

lysyjan committed:
Updated test assertions to reflect renaming of "Basic" category to "Newsletter" in email pattern categories

lysyjan committed:
Add cleanup for setTimeout in template selection modal

triple0t commented on packages/php/email-editor/src/Engine/Patterns/class-patterns.php:
```diff
@@ -33,6 +33,21 @@ private function register_block_pattern_categories(): void {
+				'name'        => 'welcome',
+				'label'       => _x( 'Welcome', 'Block pattern category', 'woocommerce' ),
+				'description' => __( 'A collection of welcome email layouts.', 'woocommerce' ),
+			),
```
It appears that these pattern categories are primarily needed by MailPoet. Is there a specific reason for registering them in the email editor core instead of within the MailPoet plugin?

lysyjan replied:
It was, but after some discussion, it is not relevant. So we can move them to MailPoet. I will do it now.

triple0t reviewed (approved):
Looks good to me. 
I dropped a minor note.

lysyjan committed:
Address PR review feedback

- Replace local PatternCategory type with UserPatternCategory imported from @wordpress/core-data/build-types/selectors
- Update JS changelog to document TemplateCategory type widening
- Update PHP changelog to document 'basic' to 'newsletter' category rename

lysyjan committed:
Remove redundant email pattern categories and update associated test assertions
